scarb/src/ops/resolve.rs

```diff
@@ -183,3 +167,25 @@ async fn collect_packages_from_resolve_graph(
 fn build_cfg_set(target: &Target) -> CfgSet {
     CfgSet::from_iter([Cfg::kv("target", target.kind.clone())])
 }
+
+#[tracing::instrument(skip_all, level = "debug")]
+pub fn is_cairo_version_compatible(
```

Comment:
```suggestion
fn check_cairo_version_compatibility(
```

this is not a predicate, `is_` naming is misleading here, it'd be slightly more enclosed function if it returned `Result<()>` and bailed inside